File: wear/src/main/java/xyz/klinker/messenger/adapter/view_holder/WearableConversationViewHolder.java
package xyz.klinker.messenger.adapter.view_holder;

import android.graphics.Typeface;
import android.graphics.drawable.ColorDrawable;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import de.hdodenhof.circleimageview.CircleImageView;
import xyz.klinker.messenger.R;
import xyz.klinker.messenger.activity.MessageListActivity;
import xyz.klinker.messenger.shared.data.Settings;
import xyz.klinker.messenger.shared.data.model.Conversation;

public class WearableConversationViewHolder extends RecyclerView.ViewHolder {

    public View headerBackground;
    public TextView header;
    public CircleImageView image;
    public TextView name;
    public TextView summary;
    public TextView imageLetter;
    public ImageView groupIcon;
    public View unreadIndicator;
    public CheckBox checkBox;

    public Conversation conversation;
    public int position = -1;

    public WearableConversationViewHolder(final View itemView) {
        super(itemView);

        this.position = -1;

        headerBackground = itemView.findViewById(R.id.header_background);
        header = (TextView) itemView.findViewById(R.id.header);
        image = (CircleImageView) itemView.findViewById(R.id.image);
        name = (TextView) itemView.findViewById(R.id.name);
        summary = (TextView) itemView.findViewById(R.id.summary);
        imageLetter = (TextView) itemView.findViewById(R.id.image_letter);
        groupIcon = (ImageView) itemView.findViewById(R.id.group_icon);
        unreadIndicator = itemView.findViewById(R.id.unread_indicator);
        checkBox = (CheckBox) itemView.findViewById(R.id.checkbox);

        if (conversation != null) {
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    MessageListActivity.startActivity(itemView.getContext(), conversation.id);
                }
            });
        }
    }


    public boolean isBold() {
        return name.getTypeface() != null && name.getTypeface().isBold();
    }

    public boolean isItalic() {
        return name.getTypeface() != null && name.getTypeface().getStyle() == Typeface.ITALIC;
    }

    public void setTypeface(boolean bold, boolean italic) {
        if (bold) {
            name.setTypeface(Typeface.DEFAULT_BOLD, italic ? Typeface.ITALIC : Typeface.NORMAL);
            summary.setTypeface(Typeface.DEFAULT_BOLD, italic ? Typeface.ITALIC : Typeface.NORMAL);

            if (unreadIndicator != null) {
                unreadIndicator.setVisibility(View.VISIBLE);
            }

            ((CircleImageView) unreadIndicator).setImageDrawable(new ColorDrawable(Settings.get(itemView.getContext()).globalColorSet.color));
        } else {
            name.setTypeface(Typeface.DEFAULT, italic ? Typeface.ITALIC : Typeface.NORMAL);
            summary.setTypeface(Typeface.DEFAULT, italic ? Typeface.ITALIC : Typeface.NORMAL);

            if (unreadIndicator != null) {
                unreadIndicator.setVisibility(View.GONE);
            }
        }
    }
}

File: wear/src/main/java/xyz/klinker/messenger/adapter/view_holder/WearableConversationViewHolder.kt
package xyz.klinker.messenger.adapter.view_holder

import android.graphics.Typeface
import android.graphics.drawable.ColorDrawable
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import de.hdodenhof.circleimageview.CircleImageView
import xyz.klinker.messenger.R
import xyz.klinker.messenger.activity.MessageListActivity
import xyz.klinker.messenger.shared.data.Settings
import xyz.klinker.messenger.shared.data.model.Conversation

class WearableConversationViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    val header: TextView by lazy { itemView.findViewById<View>(R.id.header) as TextView }
    val image: CircleImageView? by lazy { itemView.findViewById<View>(R.id.image) as CircleImageView? }
    val name: TextView by lazy { itemView.findViewById<View>(R.id.name) as TextView }
    val summary: TextView by lazy { itemView.findViewById<View>(R.id.summary) as TextView }
    val imageLetter: TextView by lazy { itemView.findViewById<View>(R.id.image_letter) as TextView }
    val groupIcon: ImageView by lazy { itemView.findViewById<View>(R.id.group_icon) as ImageView }
    private val unreadIndicator: View? by lazy { itemView.findViewById<View>(R.id.unread_indicator) }

    var conversation: Conversation? = null

    val isBold: Boolean
        get() = name.typeface != null && name.typeface.isBold

    val isItalic: Boolean
        get() = name.typeface != null && name.typeface.style == Typeface.ITALIC

    init {
        itemView.setOnClickListener {
            if (conversation != null) {
                MessageListActivity.startActivity(itemView.context, conversation!!.id)

                if (unreadIndicator != null && unreadIndicator!!.visibility == View.VISIBLE) {
                    setTypeface(false, isItalic)
                }
            }
        }
    }

    fun setTypeface(bold: Boolean, italic: Boolean) {
        if (bold) {
            name.setTypeface(Typeface.DEFAULT_BOLD, if (italic) Typeface.ITALIC else Typeface.NORMAL)
            summary.setTypeface(Typeface.DEFAULT_BOLD, if (italic) Typeface.ITALIC else Typeface.NORMAL)

            if (unreadIndicator != null) {
                unreadIndicator!!.visibility = View.VISIBLE
            }

            (unreadIndicator as CircleImageView).setImageDrawable(ColorDrawable(Settings.mainColorSet.color))
        } else {
            name.setTypeface(Typeface.DEFAULT, if (italic) Typeface.ITALIC else Typeface.NORMAL)
            summary.setTypeface(Typeface.DEFAULT, if (italic) Typeface.ITALIC else Typeface.NORMAL)

            if (unreadIndicator != null) {
                unreadIndicator!!.visibility = View.GONE
            }
        }
    }
}
